internal/k8s/syncer.go

```diff
@@ -16,24 +16,36 @@ package k8s
 import (
 	"fmt"
 
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/client-go/tools/cache"
 )
 
 // InformerSyncList holds the functions to call to check that an informer is synced.
 type InformerSyncList struct {
 	syncers []cache.InformerSynced
+	Clients *Clients
 }
 
 // InformOnResources creates informers for each of the given resources and registers their sync callbacks.
-func (sl *InformerSyncList) InformOnResources(f InformerFactory, handler *DynamicClientHandler, resources ...schema.GroupVersionResource) {
+func (sl *InformerSyncList) InformOnResources(f InformerFactory, handler *DynamicClientHandler, apiResourceList []*metav1.APIResourceList, resources ...schema.GroupVersionResource) []error {
+	var errors []error
 
 	for _, r := range resources {
+		exists, _ := sl.Clients.ResourceKindExists(r.GroupVersion().String(), r.Resource, apiResourceList)
+
+		if !exists {
+			errors = append(errors, fmt.Errorf("GroupVersion: %s Kind: %s not found in ServerGroupsAndResources", r.GroupVersion().String(), r.Resource))
+			continue
+		}
 		informer := f.ForResource(r).Informer()
 		informer.AddEventHandler(handler)
 
 		sl.syncers = append(sl.syncers, informer.HasSynced)
 	}
+
+	return errors
```

Comment:
Using errors to communicate non-error status doesn't feel that clean to me ... I had a suggestion for a different approach above. I agree with you that we we should log when we filter out a resource though, and that makes this harder to manage.

Reply:
Yeah I didn't want to plumb in all the extra meta data to properly log and thought the caller should handle the logging. I'll look at your other suggestion.